crates/bevy_scene/src/lib.rs

```diff
@@ -59,6 +59,37 @@ impl Plugin for ScenePlugin {
             .add_event::<SceneInstanceReady>()
             .init_resource::<SceneSpawner>()
             .add_systems(SpawnScene, (scene_spawner, scene_spawner_system).chain());
+
+        // Register component hooks for DynamicScene
+        app.world
+            .register_component_hooks::<Handle<DynamicScene>>()
+            .on_remove(|mut world, entity, _| {
+                let Some(handle) = world.get::<Handle<DynamicScene>>(entity) else {
+                    return;
+                };
+                let id = handle.id();
+                if let Some(&SceneInstance(scene_instance)) = world.get::<SceneInstance>(entity) {
+                    let Some(mut scene_spawner) = world.get_resource_mut::<SceneSpawner>() else {
```

Comment:
Extremely minor nit, but is there a reason that one of these is `if let` and the other one is `let ... else`, instead of both being `let ... else`?

Reply:
Looks like using `let ... else` consistently would be better. The reason for the inconsistency between the two is that in a later commit, I changed `resource_mut` to `get_resource_mut`, but I didn't notice that the `if let` expression needed to correspond to it.